build/lib/locFunc.ts

```diff
@@ -44,3 +52,301 @@ export function packageSingleExtensionStream(name : string): NodeJS.ReadWriteStr
+			let translationDataFolder = path.join(locExtFolder, 'translations');
+			if (languageId === "zh-cn") {
+				languageId = "zh-hans";
+			}
+			if (languageId === "zh-tw") {
+				languageId = "zh-hant";
+			}
+
+			//remove extensions not part of ADS.
+			if (fs.existsSync(translationDataFolder)) {
+				let totalExtensions = fs.readdirSync(path.join(translationDataFolder, 'extensions'));
+				for (let extensionTag in totalExtensions) {
+					let extensionFileName = totalExtensions[extensionTag];
+					let xlfPath = path.join(location, `${languageId}`, extensionFileName.replace('.i18n.json', '.xlf'))
+					if (!(fs.existsSync(xlfPath) || VSCODEExtensions.indexOf(extensionFileName.replace('.i18n.json', '')) !== -1)) {
+						let filePath = path.join(translationDataFolder, 'extensions', extensionFileName);
+						rimraf.sync(filePath);
+					}
+				}
+			}
+
+
+			console.log(`Importing translations for ${languageId} from '${location}' to '${translationDataFolder}' ...`);
+			let translationPaths: any = [];
+			gulp.src(path.join(location, languageId, '**', '*.xlf'))
+				.pipe(modifyI18nPackFiles(translationDataFolder, translationPaths, languageId === 'ps'))
+				.on('error', (error: any) => {
+					console.log(`Error occurred while importing translations:`);
+					translationPaths = undefined;
+					if (Array.isArray(error)) {
+						error.forEach(console.log);
+					} else if (error) {
+						console.log(error);
+					} else {
+						console.log('Unknown error');
+					}
+				})
+				.pipe(vfs.dest(translationDataFolder))
+				.on('end', function () {
+					if (translationPaths !== undefined) {
+						let nonExistantExtensions = [];
+						for (let curr of localization.translations) {
+							try {
+								if (curr.id === 'vscode.theme-seti') {
+									//handle edge case where 'theme-seti' has a different id.
+									curr.id = 'vscode.vscode-theme-seti';
+								}
+								fs.statSync(path.join(translationDataFolder, curr.path.replace('./translations', '')));
+							}
+							catch {
+								nonExistantExtensions.push(curr);
+							}
+						}
+						for (let nonExt of nonExistantExtensions) {
+							let index = localization.translations.indexOf(nonExt);
+							if (index > -1) {
+								localization.translations.splice(index, 1);
+							}
+						}
+						for (let tp of translationPaths) {
+							let finalPath = `./translations/${tp.resourceName}`;
+							let isFound = false;
+							for (let i = 0; i < localization.translations.length; i++) {
+								if (localization.translations[i].path === finalPath) {
+									localization.translations[i].id = tp.id;
+									isFound = true;
+									break;
+								}
+							}
+							if (!isFound) {
+								localization.translations.push({ id: tp.id, path: finalPath });
+							}
+						}
+						fs.writeFileSync(path.join(locExtFolder, 'package.json'), JSON.stringify(packageJSON, null, '\t'));
+					}
+				});
+
+		});
+	}
+	return new Promise(function (resolve) {
```

Comment:
What's this for? You should be able to just do something like 

```
}
console.log(...)
return Promise.resolve()
```

(assuming you change the return type as suggested above)